cpp/include/kvikio/posix_io.hpp

```diff
@@ -28,6 +28,24 @@
 
 namespace kvikio {
 
+/**
+ * @brief Type of the IO operation.
+ *
+ */
+enum class IOType : uint8_t {
+  READ,   ///< POSIX read.
+  WRITE,  ///< POSIX write.
+};
+
+/**
+ * @brief The degree to which the requested size of data is completed.
+ *
+ */
+enum class IODataCompletionLevel : uint8_t {
+  PARTIAL,  ///< POSIX read/write is called only once, which may not process all data requested.
+  FULL,     ///< POSIX read/write is called repeatedly until all requested data are processed.
+};
+
```

Comment:
I wonder if, rather than having two enums, it makes more sense to spell out the Cartesian product in the `IOType` enum:

```
enum class IOType : uint8_t {
   READ_PARTIAL,
   READ_FULL,
   WRITE_PARTIAL,
   WRITE_FULL,
};
```

If not, I'm not sure I love `IODataCompletionLevel` as a name. How about:

- `PartialIO::YES/NO`
- `IsPartial::YES/NO`

?

Reply:
Thanks for the suggestion! The first 4-choice enum looks very succinct. I'll make the changes.

Reply:
On a second thought, I probably should keep the two enums and rename the 2nd one. There's some case like [this](https://github.com/rapidsai/kvikio/pull/493/files#diff-ce16f202f49a0591c9c773df9a234a0f2198885e23ffb29d49e1e466f4d9bcedR267) where callers only need to specify read/write, and not partial/full.

Reply:
Completed.